pkg/ebpf/c/tc.v6egress.bpf.c

```diff
@@ -119,119 +115,109 @@ int handle_egress(struct __sk_buff *skb)
+				 && reverse_flow_val->addr.in6_u.u6_addr8[14] == flow_key.daddr.in6_u.u6_addr8[14]
+				 && reverse_flow_val->addr.in6_u.u6_addr8[15] == flow_key.daddr.in6_u.u6_addr8[15])) {
+		return BPF_OK;	
+	}
 
-		//Check if it's a response packet
-		reverse_flow_val = (struct conntrack_value *)bpf_map_lookup_elem(&aws_conntrack_map, &reverse_flow_key);
-		if (reverse_flow_val != NULL &&(reverse_flow_val->val[12] == src_ip[12] && reverse_flow_val->val[13] == src_ip[13]
-					&& reverse_flow_val->val[14] == src_ip[14] && reverse_flow_val->val[15] == src_ip[15])) {
-			return BPF_OK;
-		}
+	//Check if it's in the allowed list
+	trie_val = bpf_map_lookup_elem(&egress_map, &trie_key);
+	if (trie_val == NULL) {
+		bpf_ringbuf_output(&policy_events, &evt, sizeof(evt), 0);
+		return BPF_DROP;
+	}
 
-		//Check if it's in the allowed list
-		trie_val = bpf_map_lookup_elem(&egress_map, &trie_key);
-		if (trie_val == NULL) {
-			evt.verdict = 0;
+	for (int i=0; i<4; i++, trie_val++){
+		if (trie_val->protocol == RESERVED_IP_PROTOCOL) {
 			bpf_ringbuf_output(&policy_events, &evt, sizeof(evt), 0);
 			return BPF_DROP;
 		}
 
-		for (int i=0; i<4; i++, trie_val++){
-			if (trie_val->protocol == RESERVED_IP_PROTOCOL) {
-				evt.verdict = 0;
-				bpf_ringbuf_output(&policy_events, &evt, sizeof(evt), 0);
-				return BPF_DROP;
-			}
-
-			if ((trie_val->protocol == ANY_IP_PROTOCOL) || (trie_val->protocol == ip->nexthdr &&
-						((trie_val->start_port == ANY_PORT) || (l4_dst_port == trie_val->start_port) ||
-						 (l4_dst_port > trie_val->start_port && l4_dst_port <= trie_val->end_port)))) {
-				//Inject in to conntrack map
-				struct conntrack_value new_flow_val = {};
-				new_flow_val.val[0]=src_ip[0];
-				new_flow_val.val[1]=src_ip[1];
-				new_flow_val.val[2]=src_ip[2];
-				new_flow_val.val[3]=src_ip[3];
-				new_flow_val.val[4]=src_ip[4];
-                new_flow_val.val[5]=src_ip[5];
-                new_flow_val.val[6]=src_ip[6];
-                new_flow_val.val[7]=src_ip[7];
-                new_flow_val.val[8]=src_ip[8];
-                new_flow_val.val[9]=src_ip[9];
-                new_flow_val.val[10]=src_ip[10];
-                new_flow_val.val[11]=src_ip[11];
-                new_flow_val.val[12]=src_ip[12];
-                new_flow_val.val[13]=src_ip[13];
-                new_flow_val.val[14]=src_ip[14];
-                new_flow_val.val[15]=src_ip[15];
-				bpf_map_update_elem(&aws_conntrack_map, &flow_key, &new_flow_val, 0); // 0 - BPF_ANY
-				evt.verdict = 1;
-				bpf_ringbuf_output(&policy_events, &evt, sizeof(evt), 0);
-				return BPF_OK;
-			}
+		if ((trie_val->protocol == ANY_IP_PROTOCOL) || (trie_val->protocol == ip->nexthdr &&
+					((trie_val->start_port == ANY_PORT) || (l4_dst_port == trie_val->start_port) ||
+					(l4_dst_port > trie_val->start_port && l4_dst_port <= trie_val->end_port)))) {
+			//Inject in to conntrack map
+			struct conntrack_value new_flow_val;
+			__builtin_memset(&new_flow_val, 0, sizeof(new_flow_val));
```

Comment:
is there a `__builtin_bzero`

Reply:
I was reading in one the docs that bzero is getting deprecated - https://www.heatware.net/linux-tips/bzero-solution-incompatible-implicit-declaration/#

Reply:
gotcha!